src/mmwtex.c

```diff
@@ -3047,55 +3057,94 @@ void printTexLongMath(nmbrString *mathString,
     if (!g_oldTexFlag) {
       if (refType == 'e' || refType == 'f') {
         // A hypothesis - don't include \ref{}
-        printLongLine(cat("  ",
-            // If not first step, so print "\\" LaTeX line break
-            !strcmp(htmStep, "1") ? "" : "\\\\ ",
-            htmStep, // Step number
-            " && ",
-            " & ",
-            texLine,
-            // Don't put space to help prevent bad line break
-            "&\\text{Hyp~",
-            // The following puts a hypothesis number such as "2" if
-            // $e label is "abc.2"; if no ".", will be whole label.
-            right(htmRef, instr(1, htmRef, ".") + 1),
-            "}\\notag%",
-            // Add full label as LaTeX comment - note lack of space after
-            // "%" above to prevent bad line break.
-            htmRef, NULL),
-            "    \\notag \\\\ && & \\qquad ", // Continuation line prefix
-            " ");
+        texFull = cat("  ", // texFull[0] should not be a "{" character.
+          // If not first step, so print "\\" LaTeX line break
+          !strcmp(htmStep, "1") ? "" : "\\\\ ",
+          htmStep, // Step number
+          " && ",
+          " & ",
+          texLine,
+          // Don't put space to help prevent bad line break
+          "&\\text{Hyp~",
+          // The following puts a hypothesis number such as "2" if
+          // $e label is "abc.2"; if no ".", will be whole label.
+          right(htmRef, instr(1, htmRef, ".") + 1),
+          "}\\notag%",
+          // Add full label as LaTeX comment - note lack of space after
+          // "%" above to prevent bad line break.
+          htmRef, NULL),
+
+        // To avoid generating incorrect TeX, line breaking is forbidden inside
+        // scopes of curly braces.  However breaking between '\{' and '\}' is
+        // allowed.
+        // The spaces that should not be matched with 'breakMatch' are
+        // temporarily changed to ASCII 3 before the 'printLongLine' procedure
+        // is called.  The procedure rewraps the 'line' argument matching the
+        // unchanged spaces only, thus ensuring bad breaks will be avoided.
+        // The reverse is done in the 'print2()' function, where all ASCII 3's
+        // are converted back to spaces.
+        // k counts the scope level we are in.
+        k = 0;
+        n = (long)strlen(texFull);
+        for (j = 1; j < n; j++) { // We don't need to check texFull[0].
```

Comment:
Here I wasn't sure what apporach would be best. I defined the `texFull` string as the line that willl be feeded to the `printLongLine` function. The checking cicle starts at the second character of `texFull` to avoid checking the unorthodox `texFull[-1] != '\\'`. The first character is `texFull[0] = " "` so we don't need to check it since it's fixed.

Reply:
Looks good.